execution-core/src/stake.rs

```diff
@@ -282,15 +287,16 @@ pub struct StakeKeys {
     pub funds: BlsPublicKey,
 }
 
-impl StakeKeys {
-    /// Return the `MultisigPublicKey` associated to this `StakeKeys`
-    ///
-    /// # Errors
-    ///
-    /// Look at `MultisigPublicKey::aggregate`
-    pub fn multisig_pk(&self) -> Result<BlsMultisigPublicKey, BlsError> {
-        BlsMultisigPublicKey::aggregate(&[self.account, self.funds])
-    }
+/// Signature used to handle funds (stake/unstake/withdraw)
+#[derive(
+    Debug, Default, Clone, Copy, PartialEq, Eq, Archive, Deserialize, Serialize,
+)]
+#[archive_attr(derive(CheckBytes))]
+pub struct DoubleSignature {
+    /// Signature created with the account key
+    pub account: BlsSignature,
+    /// Signature created with the funds key
+    pub funds: BlsSignature,
```

Comment:
Hey, I'm curious what's the purpose of these two signatures (`account`/`funds`) if they are currently always derived from the same key anyways?

Is it just for future proofing?

Reply:
https://github.com/dusk-network/rusk/pull/2312

Reply:
Thanks!